jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/SerializedInvoker.java

```diff
@@ -45,8 +45,37 @@
     public Runnable offer(Runnable task)
     {
         if (task == null)
+        {
+            if (LOG.isDebugEnabled())
+                LOG.debug("Offering task null, skipping it in {}", this);
             return null;
+        }
+        // The NamedRunnable logger is checked to make it possible to enable the nice task names in a debugger
+        // while not flooding the logs nor stderr with logging statements just by adding
+        // org.eclipse.jetty.util.thread.SerializedInvoker$NamedRunnable.LEVEL=DEBUG to jetty-logging.properties.
+        if (NamedRunnable.LOG.isDebugEnabled())
+        {
+            if (!(task instanceof NamedRunnable))
+            {
+                // Wrap the given task with another one that's going to delegate run() to the wrapped task while the
+                // wrapper's toString() returns a description of the place in code where SerializedInvoker.run() was called.
+                StackTraceElement[] stackTrace = new Exception().getStackTrace();
+                for (StackTraceElement stackTraceElement : stackTrace)
+                {
+                    String className = stackTraceElement.getClassName();
+                    if (!className.equals(SerializedInvoker.class.getName()) && !className.equals(getClass().getName()))
```

Comment:
Can this be factored out into a `protected String deriveTaskName(Runnable task)` method?   It will make the code more readable and perhaps in some situations there is a cheaper way of naming tasks?

Reply:
isn't there something in the classname that would be cheaper than a stack?

Reply:
I could add a protected method, but wouldn't that be strange that it would only be called when debug logs are enabled? Let me try adding it, we'll see how that goes.

And there's nothing cheaper that can be done automatically AFAICT: I must walk the stack to be able to find the caller's name in first frame that isn't the invoker itself.